### DISCUSSION TITLE
Do not throw an error when dynamically branching over empty target

### DISCUSSION THREAD

**Discussion by psychelzh** [ropensci/targets#349] [Ideas]
Sometimes, the targets to be dynamically branching over can be expected to be `NULL` (i.e., empty). Currently, an error is thrown upon this, which halts the whole workflow. I am looking forward to a warning/message but not an error. For example, branching over empty target just returns an empty target (say, `NULL`) with an warning/message.

**Comment by wlandau** [comment]
This special exception would require careful work to implement, and I do not think it makes sense to branch over an empty object anyway. If the object is empty, there is no need for branching at all. You can remove it from `pattern = map(...)` and supply the whole object directly to the command.

**Reply by wlandau** [reply]
So I plan to keep the existing error condition in `targets`.

**Reply by psychelzh** [reply]
Thanks. Maybe it is okay, then. For this edge case, we had better handle it manually.
